build-ps/build-binary.sh

```diff
@@ -560,6 +560,91 @@ fi
 
 ) || exit 1
 
+# Patch needed libraries
+(
+    cd "$TARGETDIR/usr/local/$PRODUCT_FULL_NAME"
+    if [ ! -d lib/private ]; then
+        mkdir -p lib/private
+    fi
+
+    LIBLIST="libcrypto.so libssl.so libgcrypt.so libreadline.so libtinfo.so libsasl2.so libcurl.so libldap liblber libssh libbrotlidec.so libbrotlicommon.so libgssapi_krb5.so libkrb5.so libkrb5support.so libk5crypto.so librtmp.so libgssapi.so libcrypt.so libfreebl3.so libssl3.so libsmime3.so libnss3.so libnssutil3.so libplds4.so libplc4.so libnspr4.so libssl3.so libplds4.so libtirpc.so"
+    DIRLIST="bin bin/pxc_extra/pxb-8.0/bin bin/pxc_extra/pxb-2.4/bin lib bin/pxc_extra/pxb-8.0/lib/plugin bin/pxc_extra/pxb-2.4/lib/plugin lib/private lib/plugin"
+
+    LIBPATH=""
+
+    function gather_libs {
+        for lib in $LIBLIST; do
+            for elf in $(find $1 -maxdepth 1 -exec file {} \; | grep 'ELF ' | cut -d':' -f1); do
```

Comment:
It will add all executable files even if they aren't an ELF, like `./mysql-test/suite/` directory

For example, ldd will work on this output and will throw following warning:
```
root@ubuntu-20:/mnt/Percona-XtraDB-Cluster_8.0.19-10.1_Linux.x86_64.glibc2.12# ldd mysql-test/suite/tokudb/t/mvcc-27.test
	not a dynamic executable
```